_posts/2025-05-07-release-notes-3.7.0.md

```diff
@@ -0,0 +1,458 @@
+Some of the latest improvements to the presentation compiler focus on the ability to infer more information about expected types of expressions provided by the users. As the results presented compiler can now show the users the expected type of expression when hovering over holes.
+
+```scala
+def someMethod(count: Int, label: String): Unit = ???
+someMethod(???, ???)
+```
+
+Given the code snippet above hovering on the first placeholder would now provide information for Metals that `Int` is required, similarlly hovering on second `???` placehold would result in hint about missing argument of `String` type.
+
+### Quotes API changes
+
+Scala 3.7 introduces changes to the experimental subset of Quotes API. One notable addition is the `apply` methods to import selectors ([#22457](https://github.com/scala/scala3/pull/22457)), simplifying the construction of import statements within macros. This feature streamlines the generation of import trees, making macro code more concise and readable.​
+
+Another advancement is the experimental `summonIgnoring` method ([#22417](https://github.com/scala/scala3/pull/22417)). It allows developers to summon implicit instances while excluding specific instances of givens from the search. It's particularly useful in complex derivation scenarios where certain implicit should be disregarded to prevent conflicts or unintended resolutions.​ It might be used to provide a custom implementation of macro only in cases where a user-defined variant of implicit is defined.
+
+```scala
+//> using options -experimental
+
+import scala.quoted.*
+
+trait ExcludedType
+trait Show[T]:
+  def show(): Unit
+
+object Show {
+  implicit transparent inline def auto[T]: Show[T] = ${ autoImpl[T] }
+
+  private def autoImpl[T: Type](using Quotes): Expr[Show[T]] =
+    import quotes.reflect.*
+    Expr.summonIgnoring[Show[ExcludedType]](
+      // Ignore implicits produced by these symbols:
+      Symbol.classSymbol("Show").companionModule.methodMember("auto")*
+    ) match
+      case Some(instance) =>
+        // User provided their own instance of `Show[ExcludedType]` - use it 
+        '{
+          new Show[T]:
+            def show(): Unit =
+              println(s"Show[${${ Expr(TypeRepr.of[T].show) }}] including user defined Show[ExcludedType]:")
+              $instance.show()
+        }
+      case None =>
+        // No user-defined implicit for `Show[ExcludedType]` beside `Show.auto`
+        '{
+          new Show[T]:
+            def show(): Unit =
+              println(s"Show[${${ Expr(TypeRepr.of[T].show) }}]")
+        }
+}
+```
+
+Furthermore, the `Quotes.newClass` method has been extended to support class parameters, flags, privateWithin, and annotations ([#21880](https://github.com/scala/scala3/pull/21880)). This enhancement enables the dynamic creation of classes with constructors, access modifiers, and annotations, providing greater control over generated code structures.
```

Comment:
```suggestion
Furthermore, the experimental `Symbol.newClass` method has been extended to support class parameters, flags, privateWithin, and annotations ([#21880](https://github.com/scala/scala3/pull/21880)). This enhancement enables the dynamic creation of classes with constructors, access modifiers, and annotations, providing greater control over generated code structures. With those changes included, we are planning to stabilize this method (and `ClassDef.apply`) soon.
```
This isn't strictly necessary, and I don't know if we usually include future plans in the release notes, but I think it would be worth adding, as moving toward stabilization was the main goal behind including these changes now.